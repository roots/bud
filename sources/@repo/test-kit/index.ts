import type {Bud, Context} from '@roots/bud-framework'

import {path} from '@repo/constants'
import {factory} from '@roots/bud'

export type Options = {
  extensions?: Array<string>
} & Omit<Partial<Context>, `extensions`>

const makeTestBud = async (options: Options = {}): Promise<Bud> =>
  await factory({
    basedir: options.basedir ?? path(`tests`, `util`, `project`),
    cache: false,
    dry: true,
    force: true,
    mode: `production`,
    notify: false,
    silent: true,
    ...options,
  })

export type {Bud}
export {makeTestBud as factory, makeTestBud}
