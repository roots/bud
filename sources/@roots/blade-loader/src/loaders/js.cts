export default function (source: string) {
  return [...source.matchAll(/@js(?<content>[\s\S]*?)@endjs/g)]
    .map(match => match.groups.content)
    .filter(Boolean)
    .join(`\n`)
}
