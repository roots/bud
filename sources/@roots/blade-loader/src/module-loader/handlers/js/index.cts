import {extract} from '@roots/blade-loader/module-loader/handlers'

export const extension = `js`
export const pattern =
  /@module\(['"]js['"]\)(?<content>[\s\S]*?)@endmodule/g

export default (source: string) => extract(source, pattern)
