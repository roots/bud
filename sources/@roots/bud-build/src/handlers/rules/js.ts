import type {Factory} from '../index.js'

export const js: Factory = ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setUse([])
