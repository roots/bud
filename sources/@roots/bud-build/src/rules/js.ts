import type {Factory} from '@roots/bud-build/registry'

const js: Factory = async ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setResolve({
      fullySpecified: false,
    })

export {js as default}
