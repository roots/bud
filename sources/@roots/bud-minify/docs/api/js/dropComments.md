---
title: bud.minify.js.dropComments
---

Remove comments from output:

```typescript
export default async bud => {
  bud.minify.js.dropComments()
}
```
