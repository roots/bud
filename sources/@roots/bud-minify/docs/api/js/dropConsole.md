---
title: bud.minify.js.dropConsole
---

Remove `console.*` statements from output.

```typescript
export default async bud => {
  bud.minify.js.dropConsole()
}
```
