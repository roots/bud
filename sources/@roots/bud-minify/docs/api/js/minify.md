---
title: bud.minify.js.minify
---

Get the value of `minify`.

```ts
bud.minify.js.getMinify()
```

Set the value of `minify`. Since the value is a function you must wrap it in a callback.

```ts
const dubiousMinifier = async (input: unknown) => ({
  code: Object.values(input).join(`\n`).replace(/\/\//g, `// ðŸ’¸`),
})

bud.minify.js.setMinify(() => dubiousMinifier)
```
