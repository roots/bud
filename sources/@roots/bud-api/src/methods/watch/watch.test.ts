import {Bud, factory} from '@repo/test-kit'
import {beforeEach, describe, expect, it, vi} from 'vitest'

import {watch as subject} from './index.js'

describe(`bud.watch`, () => {
  describe(`in development`, () => {
    let bud: Bud
    let watch: subject

    beforeEach(async () => {
      bud = await factory({mode: `development`})
      watch = subject.bind(bud)
    })

    it(`adds watch files`, async () => {
      await watch(`1/*.js`)

      const value = bud.hooks.filter(`dev.watch.files`)

      if (!(value instanceof Set))
        throw new Error(`watch files should be a set`)

      expect(Array.from(value)).toMatchSnapshot(
        expect.arrayContaining([`1/*.js`]),
      )
    })

    it(`adds files even if none have ever been added`, async () => {
      let result

      // @ts-ignore
      bud.hooks.on = vi.fn((key: any, hookFn: any) => {
        result = hookFn(undefined)
        return bud
      })

      await watch(`1/*.js`)

      // @ts-ignore
      expect(Array.from(result)).toEqual(
        expect.arrayContaining([`1/*.js`]),
      )
    })
  })

  describe(`in production`, () => {
    let bud: Bud
    let watch: subject

    beforeEach(async () => {
      bud = await factory({mode: `production`})
      watch = subject.bind(bud)
    })

    it(`should not run`, async () => {
      expect(bud.mode).toBe(`production`)

      const watch = subject.bind(bud)

      await watch(`1/*.js`)

      const value = bud.hooks.filter(`dev.watch.files`)
      expect(value).toBeUndefined()
    })
  })
})
