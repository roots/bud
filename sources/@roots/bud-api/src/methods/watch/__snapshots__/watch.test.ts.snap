// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bud.watch > in development > adds watch files 1`] = `
[
  "1/*.js",
]
`;
